spec/OpenLineage.yml

```diff
@@ -0,0 +1,250 @@
+openapi: 3.0.2
+info:
+  title: OpenLineage
+  version: 0.0.1
+  description: OpenLineage is an open source **lineage and metadata collection API** for the data ecosystem.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+paths:
+  /events:
+    post:
+      summary: Updates the state of a run
+      description: Updates a run state for a job.
+      operationId: postRunStateUpdate
+      tags:
+        - OpenLineage
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RunStateUpdates'
+      responses:
+        '200':
+          description: OK
+
+components:
+  schemas:
+    RunStateUpdates:
+      type: array
+      items:
+        $ref: '#/components/schemas/RunStateUpdate'
+
+    RunStateUpdate:
+      type: object
+      properties:
+        runId:
+          description: the unique identifier for that run
+          type: string
+          format: uuid
+        jobId:
+          $ref: '#/components/schemas/JobId'
+        transition:
+          description: transitions the run state
+          type: string #TODO: add enum
+          example: 'START|RUNNING|COMPLETE|ABORT|FAIL'
+        transitionTime:
+          type: string
+          format: date-time
+        origin:
+          $ref: '#/components/schemas/Origin'
+        job:
+          $ref: '#/components/schemas/Job'
+        inputs:
+          description: The set of **input** datasets.
+          type: array
+          items:
+            $ref: '#/components/schemas/Dataset'
+        outputs:
+          description: The set of **output** datasets.
+          type: array
+          items:
+            $ref: '#/components/schemas/Dataset'
+        facets:
+          description: The run facets.
+          type: object
```

Comment:
Should this be an array?

Reply:
It could. Since we're using JSON and it is required for the facet name to be unique in the list, I took the approach to model it as:
```
facets: {
  "{facet_name}": { .... facet properties ...}
  ...
}
```
The array equivalent would look like this:
```
facets: [
   { name: "{facet_name}", ... facet properties ...},
   ...
]
```
I have a preference for the object approach but I'm happy to hear other opinions.